### ISSUE TITLE
Open Discussion: Use python dependencies management

### ISSUE THREAD

**Issue by arneboockmeyer** [osmhpi/cohydra#11]
I would recommend to use a python based dependencies management system. Therefore I would create a `setup.py` with details about the project and all dependencies (like `aexpr` or `pylxd`). 

See: https://packaging.python.org/tutorials/packaging-projects/
And: https://python-packaging.readthedocs.io/en/latest/dependencies.html
Example: https://github.com/active-expressions/active-expressions-static-python/blob/master/setup.py

Advantages:
- Use common way to manage python dependencies
- Let python manage his dependencies itself - and not docker
- Offers the option to have a pip installation of the testbed

**Comment by lpirl** [comment]
We should definitively provide a ``setup.py``.

Everything beyond ``pip install <repo-url>`` is too much burden for users.

**Comment by arneboockmeyer** [comment]
:+1: @lpirl I totally agree. And it would be really really nice, if we are able to provide the functionality of the testbed just with `pip install <repo-url>`

**Comment by arneboockmeyer** [comment]
Then the installation inside of the container could only be `pip install <repo-url>`. This would keep the dependencies at one place.

**Comment by mgjm** [comment]
If we manage to build an easily installable package for ns-3, then our cohydra framework can depend on this ns-3 package. But the current situation is that the workflow to get the ns-3 python package is quite tedious and requires manual recompilation. Therefore I think it is out of the scope for this project to build such an easily installable python package for ns-3.
But all other dependencies can be managed with e.g. `pip`.

But then it would still not be possible to `pip install cohydra` 🤔

**Comment by lpirl** [comment]
It is true that cohydra has dependencies beyond `setup.py` and that we probably won't focus on streamlining the on-boarding process.
From my understanding, this issue is about the Python packages which cohydra requires and which can be installed from the CheeseShop right away.

**Comment by arneboockmeyer** [comment]
@mgjm Yes, we will not build a ns-3 pip package or something like that. But we also do not install docker, lxd or sumo on the machines. But we should cover as much as possible at a single point. The `setup.py` is the intended way from python and we should follow that. I opened this issue because I tried to run the lxd branch and it crashed the first time because of a missing dependency. We should fix that.

**Comment by lpirl** [comment]
For installing cohydra with all its dependencies, we /could/ provide – for
example – an Ansible role but that's another discussion (just to put the idea
into heads here).
